preprocessing/pipeline.py

```diff
@@ -1,10 +1,15 @@
 from concurrent.futures import ProcessPoolExecutor
+from image_processor_interface import ImageProcessor
+from skip_image import SkipImage
 
 class Pipeline:
-    def __init__(self, steps: list[IImageProcessor]):
+    def __init__(self, steps: list[ImageProcessor]):
         self.steps = steps
 
     def execute(self, img):
-        for step in self.steps:
-            img = step.process(img) # Guarantees this method exists
-        return img
+        try:
+            for step in self.steps:
+                img = step.process(img)
+                return img
```

Comment:
```suggestion
          return img
```
should be not inside the loop, otherwise it returns the first and only image

Reply:
To clarify, it returns the image after the first process function